### PR TITLE
Rewrote test_platforms to use mocking based approach.

### PR DESCRIPTION
**Pull request by adusca**
As seen in #144 most of our previous tests depended on specific buildernames being in allthethings.json. I rewrote those tests to use mock_allthethings.json instead (and deleted test_platforms.json, which is no longer used).

This will probably reduce coverage because we are no longer indirectly testing allthethings.py.

### CODE REVIEW

File: test/test_platforms.py [hunk 1]
```diff
@@ -16,136 +10,105 @@
...
+    def test_valid(self, fetch_allthethings_data):
+        """Test if the function finds the right builder."""
+        fetch_allthethings_data.return_value = MOCK_ALLTHETHINGS
+        self.assertEquals(mozci.platforms.determine_upstream_builder('Platform1 repo mochitest-1'),
+                          'Platform1 repo build')
+        self.assertEquals(
+            mozci.platforms.determine_upstream_builder('Platform1 repo debug test mochitest-1'),
+            'Platform1 repo leak test build')
+        self.assertEquals(mozci.platforms.determine_upstream_builder('Platform1 repo talos tp5o'),
+                          'Platform1 repo build')
```
Comment (armenzg) [inline test/test_platforms.py]: Could you please make these 3 asserts brackets and indentations look a bit more uniform?
(aka they open brackets around the same column)

Or make them 3 different tests? Whatever you prefer.

**Comment by armenzg** [comment]
You’re absolutely amazing. Thanks for tackling this!
I really love your attention to details. I don’t get to see this often!

Let’s just add some more test cases for issue #151.

**Comment by adusca** [comment]
The second commit breaks the build because I added a test case to catch #151. I'm working on a fix for that issue.

**Comment by adusca** [comment]
I ended up working on #151 on this branch (it was easier because of the tests). It's a little hacky but it works:

> > > from mozci.platforms import determine_upstream_builder
> > > determine_upstream_builder("Ubuntu HW 12.04 x64 mozilla-aurora talos tp5o")
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "mozci/platforms.py", line 106, in determine_upstream_builder
> > >     raise Exception('Did you mean %s' % buildername_with_pgo)
> > > Exception: Did you mean Ubuntu HW 12.04 x64 mozilla-aurora pgo talos tp5o
> > > determine_upstream_builder("Ubuntu HW 12.04 x64 mozilla-aurora pgo talos tp5o")
> > > 'Linux x86-64 mozilla-aurora build'
> > > determine_upstream_builder("Rev5 MacOSX Yosemite 10.10 mozilla-aurora talos tp5o")
> > > 'OS X 10.7 mozilla-aurora build'

**Comment by adusca** [comment]
Is it possible that both "Platform repo pgo talos test" and "Platform repo talos test" are valid for some repo? If not it's probably better to just remove the "if 'mozilla-beta' in buildername or 'mozilla-aurora' in buildername"

It would be possible to use the same logic with schedulers: 
Since both "tests-mozilla-aurora-linux64-talos" and "tests-mozilla-aurora-linux64-pgo-talos" exist, we should ignore every job in "tests-mozilla-aurora-linux64-talos", and since "mozilla-aurora-macosx64-pgo-talos" does not exist we should not ignore "mozilla-aurora-macosx64-talos". We would need to keep a list of every scheduler name.

**Comment by armenzg** [comment]
If you could use the schedulers to resolve this that would be great as I don't know what the future will hold for the configurations for each repo.

However, since things are moving to TaskCluster in the next year, I would not kill myself if it becomes convoluted. Whatever you choose is fine with me.

> Is it possible that both "Platform repo pgo talos test" and "Platform repo talos test" are valid for
> some repo? If not it's probably better to just remove the "if 'mozilla-beta' in buildername or
> 'mozilla-aurora' in buildername"

Most trunk trees (e.g. mozilla-central, mozilla-inbound) have both.
Opt builds are triggered per push.
Pgo builds are triggered every few hours (since they take much longer to run).
You will see that [1] has triggered 1 opt chromez and 2 pgo chromez.

The pgo jobs have been triggered by the pgo build and the nightly build while the opt talos was triggered by the opt build [2](two jobs from first row and job from second row)

[1] https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&filter-searchStr=chromez&revision=8af276ab8636
[2] https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&filter-searchStr=Linux%20x86-64%20mozilla-central&revision=8af276ab8636

**Comment by adusca** [comment]
Using schedulers we would also have to hardcode mozilla-aurora and mozilla-beta (because some repos have both pgo talos and talos as valid jobs), so I don't think there is an advantage.

I realized that this PR would break scripts/misc/generate_list_of_testers_without_builders.py so I add a commit to make it robust to Exceptions.

New output:
Ubuntu HW 12.04 mozilla-aurora talos chromez
Ubuntu HW 12.04 mozilla-aurora talos dromaeojs
Ubuntu HW 12.04 mozilla-aurora talos g1
Ubuntu HW 12.04 mozilla-aurora talos other_nol64
Ubuntu HW 12.04 mozilla-aurora talos svgr
Ubuntu HW 12.04 mozilla-aurora talos tp5o
Ubuntu HW 12.04 mozilla-beta talos chromez
Ubuntu HW 12.04 mozilla-beta talos dromaeojs
Ubuntu HW 12.04 mozilla-beta talos g1
Ubuntu HW 12.04 mozilla-beta talos other_nol64
Ubuntu HW 12.04 mozilla-beta talos svgr
Ubuntu HW 12.04 mozilla-beta talos tp5o
Ubuntu HW 12.04 x64 mozilla-aurora talos chromez
Ubuntu HW 12.04 x64 mozilla-aurora talos dromaeojs
Ubuntu HW 12.04 x64 mozilla-aurora talos g1
Ubuntu HW 12.04 x64 mozilla-aurora talos other_l64
Ubuntu HW 12.04 x64 mozilla-aurora talos svgr
Ubuntu HW 12.04 x64 mozilla-aurora talos tp5o
Ubuntu HW 12.04 x64 mozilla-beta talos chromez
Ubuntu HW 12.04 x64 mozilla-beta talos dromaeojs
Ubuntu HW 12.04 x64 mozilla-beta talos g1
Ubuntu HW 12.04 x64 mozilla-beta talos other_l64
Ubuntu HW 12.04 x64 mozilla-beta talos svgr
Ubuntu HW 12.04 x64 mozilla-beta talos tp5o
Windows 7 32-bit mozilla-aurora talos chromez
Windows 7 32-bit mozilla-aurora talos dromaeojs
Windows 7 32-bit mozilla-aurora talos g1
Windows 7 32-bit mozilla-aurora talos other_nol64
Windows 7 32-bit mozilla-aurora talos svgr
Windows 7 32-bit mozilla-aurora talos tp5o
Windows 7 32-bit mozilla-aurora talos xperf
Windows 7 32-bit mozilla-beta talos chromez
Windows 7 32-bit mozilla-beta talos dromaeojs
Windows 7 32-bit mozilla-beta talos g1
Windows 7 32-bit mozilla-beta talos other_nol64
Windows 7 32-bit mozilla-beta talos svgr
Windows 7 32-bit mozilla-beta talos tp5o
Windows 7 32-bit mozilla-beta talos xperf
Windows 8 64-bit mozilla-aurora talos chromez
Windows 8 64-bit mozilla-aurora talos dromaeojs
Windows 8 64-bit mozilla-aurora talos g1
Windows 8 64-bit mozilla-aurora talos other_nol64
Windows 8 64-bit mozilla-aurora talos svgr
Windows 8 64-bit mozilla-aurora talos tp5o
Windows 8 64-bit mozilla-beta talos chromez
Windows 8 64-bit mozilla-beta talos dromaeojs
Windows 8 64-bit mozilla-beta talos g1
Windows 8 64-bit mozilla-beta talos other_nol64
Windows 8 64-bit mozilla-beta talos svgr
Windows 8 64-bit mozilla-beta talos tp5o
Windows XP 32-bit mozilla-aurora talos chromez
Windows XP 32-bit mozilla-aurora talos g1
Windows XP 32-bit mozilla-aurora talos other_nol64
Windows XP 32-bit mozilla-aurora talos svgr
Windows XP 32-bit mozilla-aurora talos tp5o
Windows XP 32-bit mozilla-beta talos chromez
Windows XP 32-bit mozilla-beta talos g1
Windows XP 32-bit mozilla-beta talos other_nol64
Windows XP 32-bit mozilla-beta talos svgr
Windows XP 32-bit mozilla-beta talos tp5o

**Comment by armenzg** [comment]
OK. Sounds good.

Could we return None instead of an Exception? I forgot that is what we actually do in other cases. My apologies about this. Please feel free to merge after that.

Thanks again!

**Pull request merged**